mimic/test/test_auth.py

```diff
@@ -809,3 +819,218 @@ def test_impersonation_request_with_invalid_body_causes_http_bad_request(self):
+                        "id": "123456a"
+                    }
+                }
+            }
+        ))
+        self.assertEqual(response.code, 200)
+        self.assertEqual(json_body["access"]["token"]["id"],
+                         json_body1["access"]["token"]["id"])
+        self.assertEqual(json_body["access"]["token"]["tenant"]["id"],
+                         json_body1["access"]["token"]["tenant"]["id"])
+        self.assertEqual(json_body["access"]["user"]["name"],
+                         json_body1["access"]["user"]["name"])
+
+    def test_response_for_validate_impersonated_token(self):
+        """
+        Test to verify :func: `validate_token` and then authenticate
+        """
+        core = MimicCore(Clock(), [ExampleAPI()])
+        root = MimicRoot(core).app.resource()
+        # Authenticate the impersonator (admin user)
+        (response0, json_body0) = self.successResultOf(json_request(
+            self, root, "POST", "/identity/v2.0/tokens",
+            {
+                "auth": {
+                    "tenantId": "111111",
+                    "token": {
+                        "id": "123456a"
+                    }
+                }
+            }
+        ))
+        self.assertEqual(200, response0.code)
+        impersonator_token = json_body0["access"]["token"]["id"]
+
+        # Authenticate using the username so we know the tenant_id
+        (response1, json_body1) = self.successResultOf(json_request(
+            self, root, "POST", "/identity/v2.0/tokens",
+            {
+                "auth": {
+                    "passwordCredentials": {
+                        "username": "test1",
+                        "password": "theUsersPassword"
+                    },
+                    "tenantId": "12345"
+                }
+            }
+        ))
+        self.assertEqual(200, response1.code)
+
+        # Impersonate user test1
+        (response2, json_body2) = self.successResultOf(json_request(
+            self, root, "POST",
+            "http://mybase/identity/v2.0/RAX-AUTH/impersonation-tokens",
+            {"RAX-AUTH:impersonation": {"expire-in-seconds": 1,
+                                        "user": {"username": "test1"}}},
+            headers={'X-Auth-Token': [str(impersonator_token)]}
+        ))
+        self.assertEqual(200, response2.code)
+        impersonated_token = json_body2["access"]["token"]["id"]
+        # validate the impersonated_token
+        (response3, json_body3) = self.successResultOf(json_request(
+            self, root, "GET",
+            "http://mybase/identity/v2.0/tokens/{0}?belongsTo=12345".format(
+                impersonated_token)
+        ))
+        self.assertEqual(200, response3.code)
+        self.assertTrue(json_body3["access"]["RAX-AUTH:impersonator"])
+
+    def test_response_for_validate_impersonated_token_multiple_users(self):
+        """
+        Test to verify :func: `validate_token` and then authenticate
+        """
```

Comment:
This test method is reaaaaaaaally long; I notice that there are a lot of lines which are basically just repeated with different values.  Could you put some of this stuff into a function, maybe even just an inner function to start with, to cut down on the length?